drivers/regulator/s2mps11.c

```diff
@@ -1244,6 +1244,31 @@ static int s2mps11_pmic_probe(struct platform_device *pdev)
 	return ret;
 }
 
+static void s2mps11_pmic_shutdown(struct platform_device *pdev)
```

Comment:
this is another patch that has been rolled on since forever.
afaik, there are some issues with the ethernet adapter (on XU3 or XU4, can't remember) where sometimes randomly  after reboot the device disappears - a quick solution was to reset it here (turn S2MPS11_REG_L15CTRL and S2MPS11_REG_L17CTRL off and on).